lib/framework/data_context.dart

```diff
@@ -546,45 +546,52 @@ class EnsembleStorage with Invokable {
   }
 }
 
-/// singleton handling utils
 class EnsembleUtils with Invokable {
-  static final EnsembleUtils _obj = EnsembleUtils._();
-  EnsembleUtils._();
-  factory EnsembleUtils() => _obj;
-
   @override
-  Map<String, Function> getters() => {'getCountries': () => allCountries};
+  Map<String, Function> getters() => {};
 
   @override
   Map<String, Function> methods() => {
-        'getCountries': (value) {
+        'getCountries': () => allCountries,
+        'getCountry': (value) {
           String val = Utils.getString(value, fallback: "");
-          return CountryManager().getCountryData(val);
+          return getCountry(val);
         },
+        'findCountry': (value) {
+          String val = Utils.getString(value, fallback: "");
+          return findCountry(val);
+        }
       };
 
   @override
   Map<String, Function> setters() => {};
-}
 
-class CountryManager {
-  static final CountryManager _instance = CountryManager._();
-  CountryManager._();
-  factory CountryManager() => _instance;
+  List<Map<String, dynamic>> getCountry(String val) {
+    String input = val.toLowerCase().trim();
+    List<Map<String, dynamic>> alpha2List = [];
+    for (var i in allCountries) {
+      String countryCode = i['iso']['alpha-2'];
+      countryCode = countryCode.toLowerCase();
+      if (countryCode == input) {
+        alpha2List.add(i);
+      }
+    }
+    return alpha2List;
+  }
 
-  List<Map<String, dynamic>> getCountryData(String userInput) {
+  List<Map<String, dynamic>> findCountry(String userInput) {
```

Comment:
Problems with this function:
- you are making 2 passes through the list. We should only do it once for performance reason.
- By concating the list, you essentially break the existing alphabetical sort order. 

Please use 1 pass only and keep the original JSON sort order.